Tests/scripts/validate_files_structure.py

```diff
@@ -682,6 +703,222 @@ def str2bool(v):
+
+                # if detected disable-secrets comment, skip the line
+                if bool(re.findall(r'(disable-secrets-detection)', line)):
+                    file_contents = file_contents.replace(line, '')
+                    continue
+
+                # REGEX scanning for IOCs and false positive groups
+                potential_secrets, false_positives = regex_for_secrets(line)
+                for potential_secret in potential_secrets:
+                    if not any(white_list_string in potential_secret for white_list_string in secrets_white_list):
+                        regex_secrets.append(potential_secret)
+                # added false positives into white list array before testing the strings in line
+                secrets_white_list = secrets_white_list.union(false_positives)
+
+                # calculate entropy for each string in the file
+                for string in line.split():
+                    string = string.strip("\"()[],'><:;\\")
+                    string_lower = string.lower()
+                    # compare the lower case of the string against both generic whitelist & temp white list
+                    if not any(white_list_string in string_lower for white_list_string in secrets_white_list):
+                        entropy = calculate_shannon_entropy(string)
+                        if entropy >= ENTROPY_THRESHOLD:
+                            high_entropy_strings.append(string)
+
+        if high_entropy_strings or regex_secrets:
+            # uniquify identical matches between lists
+            file_secrets = list(set(high_entropy_strings + regex_secrets))
+            secrets_found[file_name] = file_secrets
+
+    return secrets_found
+
+
+def create_temp_white_list(file_contents):
+    temp_white_list = set([])
+    context_paths = re.findall(r'contextPath: (\S+\.+\S+)', file_contents)
+    for context_path in context_paths:
+        context_path = context_path.split('.')
+        context_path = [white_item.lower() for white_item in context_path]
+        temp_white_list = temp_white_list.union(context_path)
+
+    return temp_white_list
+
+
+def retrieve_related_yml(file_path_temp):
+    matching_yml_file_contents = None
+    yml_file = file_path_temp + '.yml'
+    if os.path.exists(yml_file):
+        with io.open('./' + yml_file, mode="r", encoding="utf-8") as matching_yml_file:
+            matching_yml_file_contents = matching_yml_file.read()
+    return matching_yml_file_contents
+
+
+def regex_for_secrets(file_contents):
+    """
+    Scans for IOCs with potentially low entropy score
+    :param file_contents: file to test as string representation (string)
+    :return  potential_secrets (list) IOCs found via regex, false_positives (list) Non secrets with high entropy
+    """
+    potential_secrets = []
+    false_positives = []
+
+    # URL REGEX
+    urls = re.findall(r'http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\), ]|(?:%[0-9a-fA-F][0-9a-fA-F]))+', file_contents)
```

Comment:
Maybe create a CONST for each regex?

Reply:
I did this for readability mostly, if you think const makes for better convention I will switch